Sources/MessagingPush/MessagingPushImplementation.swift

```diff
@@ -48,6 +31,9 @@ class MessagingPushImplementation: MessagingPushInstance {
         event: Metric,
         deviceToken: String
     ) {
-        richPushDeliveryTracker.trackMetric(token: deviceToken, event: event, deliveryId: deliveryID) { _ in }
+        // Access richPushDeliveryTracker from DIGraphShared.shared directly as it is only required for NSE.
+        // Keeping it as class property results in initialization of UserAgentUtil before SDK client is overridden by wrapper SDKs.
+        // In future, we can improve how we access SdkClient so that we don't need to worry about initialization order.
+        DIGraphShared.shared.richPushDeliveryTracker.trackMetric(token: deviceToken, event: event, deliveryId: deliveryID) { _ in }
```

Comment:
We should ideally limit the exposure of this class in NSE to avoid this.

Reply:
Agreed. I avoided it to prevent scope creep for now, but you're right, the proper fix should aim to avoid similar issues in the future. Exposing these methods specifically for NSE is one good option for this. I'll create ticket to follow this up.